sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4934,6 +4934,11 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
 
     checkAnswer(df, Row("a"))
   }
+
+  test("SPARK-51805: Get with null argument should return null instead of throwing cast error") {
```

Comment:
Would you mind to move this test to more specific place, for example `array.sql`